sdl_android/src/main/java/com/smartdevicelink/api/lockscreen/LockScreenManager.java

```diff
@@ -0,0 +1,274 @@
+		if (lockScreenEnabled) {
+			LockScreenStatus status = getLockScreenStatus();
+			if (hmiLevel == HMILevel.HMI_FULL && status == LockScreenStatus.REQUIRED) {
+				Intent showLockScreenIntent = new Intent(context.get(), SDLLockScreenActivity.class);
+				showLockScreenIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+				// Extra parameters for customization of the lock screen view
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_ICON_EXTRA, lockScreenIcon);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_COLOR_EXTRA, lockScreenColor);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_CUSTOM_VIEW_EXTRA, customView);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_OEM_ICON_EXTRA, showOEMLogo);
+				showLockScreenIntent.putExtra(SDLLockScreenActivity.LOCKSCREEN_OEM_ICON_BITMAP, lockScreenOEMIcon);
+
+				context.get().startActivity(showLockScreenIntent);
+			} else if (status == LockScreenStatus.OFF) {
+				context.get().sendBroadcast(new Intent(SDLLockScreenActivity.CLOSE_LOCK_SCREEN_ACTION));
+			}
+		}
+	}
+
+	////
+	// HELPERS
+	////
+
+	/**
+	 * Step through some logic to determine if we need to show the lock screen or not
+	 * This function is usually triggered on some sort of notification.
+	 *
+	 * @return Whether or not the Lock Screen is required
+	 */
+	private synchronized LockScreenStatus getLockScreenStatus() {
+
+		if ( (hmiLevel == null) || (hmiLevel.equals(HMILevel.HMI_NONE))) {
+			return LockScreenStatus.OFF;
+		}
+		else if ( hmiLevel.equals(HMILevel.HMI_BACKGROUND)) {
+			if (!driverDistStatus) {
+				//we don't have driver distraction, lock screen is entirely based on if user is using the app on the head unit
+				return LockScreenStatus.OFF;
+			} else {
+				return LockScreenStatus.REQUIRED;
+			}
+		}
+		else if ( (hmiLevel.equals(HMILevel.HMI_FULL)) || (hmiLevel.equals(HMILevel.HMI_LIMITED))) {
+			if (!driverDistStatus) {
+				return LockScreenStatus.OPTIONAL;
+			} else {
+				return LockScreenStatus.REQUIRED;
+			}
+		}
+		return LockScreenStatus.OFF;
+	}
+
+	private void downloadLockScreenIcon(final String url){
+		new Thread(new Runnable(){
+			@Override
+			public void run(){
+				try{
+					lockScreenOEMIcon = HttpUtils.downloadImage(url);
+					Log.i(TAG, "Lock Screen Icon Downloaded");
+				}catch(IOException e){
+					Log.e(TAG, "Lock Screen Icon Error Downloading");
+				}
+			}
+		}).start();
+	}
+
+	// protected getters for testing class
+	protected int getLockScreenIcon(){
+		return lockScreenIcon;
+	}
+
+	protected int getLockScreenColor(){
+		return lockScreenColor;
+	}
+
+	protected int getCustomView(){
+		return customView;
+	}
+
+	protected boolean getShowOEMLogo(){
+		return showOEMLogo;
+	}
+
+	protected boolean getLockScreenEnabled(){
+		return lockScreenEnabled;
+	}
+
+	protected Bitmap getLockScreenOEMIcon(){
+		return lockScreenOEMIcon;
+	}
+
+	protected LockScreenStatus testGetLockScreenStatus(){
```

Comment:
Why is this method included? Can't the getLockScreenStatus method just be `protected` instead of `private` since that's what this method essentially does?